engine/modules/camera/src/main/java/com/codingame/gameengine/module/camera/CameraModule.java

```diff
@@ -0,0 +1,165 @@
+package com.codingame.gameengine.module.camera;
+
+import com.codingame.gameengine.core.AbstractPlayer;
+import com.codingame.gameengine.core.GameManager;
+import com.codingame.gameengine.core.Module;
+import com.codingame.gameengine.module.entities.Entity;
+import com.codingame.gameengine.module.entities.GraphicEntityModule;
+import com.google.inject.Inject;
+import com.google.inject.Singleton;
+
+import java.util.ArrayList;
+import java.util.Arrays;
+import java.util.HashMap;
+import java.util.Map;
+
+/**
+ * The CameraModule allow you to have a dynamic camera following a set of objects
+ */
+@Singleton
+public class CameraModule implements Module {
+
+    GameManager<AbstractPlayer> gameManager;
+    @Inject
+    GraphicEntityModule entityModule;
+    Map<Integer, Boolean> registered, newRegistration;
+    ArrayList<Entity<?>> trackedEntities;
+    Double cameraOffset;
+    Integer container, sizeX, sizeY = -1;
+    boolean sentContainer = false;
+    double previousOffset = -1;
+    boolean active = true;
+    boolean oldActive = false;
+
+
+    @Inject
+    CameraModule(GameManager<AbstractPlayer> gameManager) {
+        trackedEntities = new ArrayList<>();
+        this.gameManager = gameManager;
+        gameManager.registerModule(this);
+        registered = new HashMap<>();
+        newRegistration = new HashMap<>();
+        cameraOffset = 10.;
+    }
+
+    @Override
+    public void onGameInit() {
+        sendFrameData();
+    }
+
+    @Override
+    public void onAfterGameTurn() {
+        sendFrameData();
+    }
+
+    @Override
+    public void onAfterOnEnd() {
+    }
+
+
+    private void sendFrameData() {
+        Object[] data = {null, null, null, null};
+        Object[] empty = {null, null, null, null};
+        if (!newRegistration.isEmpty()) {
+            data[0] = new HashMap<>(newRegistration);
+            System.out.printf("added size : %d\n", newRegistration.size());
```

Comment:
Forgot to remove your debug code?

Reply:
Solved with the last commit. Thinks for pointing it out. I also cleaned up the JS code.